src/main/java/org/broadinstitute/hellbender/tools/spark/sv/RunSGANaivelySpark.java

```diff
@@ -0,0 +1,485 @@
+
+        validateAndSaveResults(assembledContigs, outputDir);
+    }
+
+    // validates the returned result:
+    //   if all steps return 0, save the contig file and may discard the runtime information
+    //   if any step returns non-zero code, save the runtime information for that break point
+    private static void validateAndSaveResults(final JavaPairRDD<Long, SGAAssemblyResultType> results, final String outputDir){
+        results.cache(); // cache because Spark doesn't have an efficient RDD.split(predicate) yet
+        final JavaPairRDD<Long, SGAAssemblyResultType> success = results.filter(entry -> entry._1()!=null);
+        success.mapToPair(entry -> new Tuple2<>(entry._1(), new ContigsCollection(entry._2().assembledContigFile)))
+                .saveAsObjectFile(outputDir);
+        final JavaPairRDD<Long, SGAAssemblyResultType> failure = results.filter(entry -> entry._2()==null);
+        failure.mapToPair(entry -> new Tuple2<>(entry._1(), entry._2().runtimeInformation))
+                .saveAsObjectFile(outputDir);
+    }
+
+    /**
+     * Converts an entry in an JavaPairRDD whose first is an breakpoint id and second is an iterable of paired reads
+     *    where either end is "mapped" (which ever way the read picker think mapping means) to a place near
+     *    the breakpoint represented by the id.
+     * @param orderedReadsOfABreakPoint  the entry to be converted
+     * @return                           a tuple corresponding to this breakpoint where the first is still the breakpoint id,
+     *                                   and the second is another tuple of the temporary directory and a temporary FASTQ file
+     *                                   containing sequence information.
+     * @throws IOException               an IOException is thrown if the conversion fails to create the FASTQ file
+     */
+    @VisibleForTesting
+    static Tuple2<Long, File> convertReadsToFASTQ(final Tuple2<Long, Iterable<GATKRead>> orderedReadsOfABreakPoint) throws IOException{
+
+        final Long breakpointId = orderedReadsOfABreakPoint._1();
+
+        File workingDir = new File( Files.createTempDirectory("assembly" + orderedReadsOfABreakPoint._1().toString()).toString() );
+        if(null==workingDir){
+            throw new IOException("Failed to create temporary directory to perform assembly in.");
+        }
+        IOUtils.deleteRecursivelyOnExit(workingDir);
+
+        final File fastq = new File(workingDir, "assembly" + breakpointId + ".fastq");
+        if(null==fastq){
+            throw new IOException("Failed to create temporary FASTQ file to feed to SGA.");
+        }
+
+        try(final FastqWriter writer = new FastqWriterFactory().newWriter(fastq)){
+            Iterator<GATKRead> it = orderedReadsOfABreakPoint._2().iterator();
+            while(it.hasNext()){
+                writeToTempFASTQ(it.next(), writer);
```

Comment:
I'd be explicit in the comments that this method expects and writes out interleaved reads (first of pair followed by second of pair in the same file). Is that true?